File: backend/quizhire/quizhire/src/main/java/com/quizhire/quizhire/services/TestResultService.java
package com.quizhire.quizhire.services;

import java.util.List;

import com.quizhire.quizhire.models.test.Test;
import com.quizhire.quizhire.models.test.TestResult;

public interface TestResultService {
    TestResult getTestResultById(Integer id) throws Exception;
    List<TestResult> getTestResultByTest(Test test) throws Exception;
    TestResult CreateTestResult(TestResult testResult);
    

}

File: frontend/quiz-plateform/src/app/services/test-result.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';
import { Observable } from 'rxjs';
import { TestResult } from '../models/testResult';

@Injectable({
  providedIn: 'root'
})
export class TestResultService {

  constructor(private http: HttpClient) { }

  getTestResultById(id: number): Observable<TestResult> {
    return this.http.get<TestResult>(`${baseUrl}/test-results/${id}`);
  }

  createTestResult(testResult: TestResult): Observable<TestResult> {
    return this.http.post<TestResult>(`${baseUrl}/test-results`, testResult);
  }

  getTestResultsByTest(testId: number): Observable<TestResult[]> {
    return this.http.get<TestResult[]>(`${baseUrl}/test-results/test/${testId}`);
  }
}